js/views/homeVw.js

```diff
@@ -449,44 +484,32 @@ module.exports = Backbone.View.extend({
     }
   },
 
-  removeSubView: function(view) {
-    var self = this,
-        index;
+  _removeSubView: function(list, view) {
+    if (list && Array.isArray(list) && view) {
+      for (var i = list.length; i--;) {
+        if (view == list[i]) {
+          view.remove();
+          break;
+        }
+      }
+    }
+  },
 
+  removeItemView: function(view) {
     if (!view) return;
 
-    __.every(this.subViews, function(subView, i) {
-      if (view === subView) {
-        index = i;
-
-        if(subView.close){
-          subView.close();
-        }else{
-          subView.unbind();
-          subView.remove();
-        }
-
-        self.subViews.splice(index, 1);
+    this._removeSubView(this.itemViews, view);
+  },
 
-        return false;        
-      }
+  removeUserView: function(view) {
+    if (!view) return;
 
-      return true;
-    });
+    this._removeSubView(this.userViews, view);
   },
 
-  close: function(){
-    __.each(this.subViews, function(subView) {
-      if(subView.close){
-        subView.close();
-      }else{
-        subView.unbind();
-        subView.remove();
-      }
-    });
-    clearInterval(this.homeLookingTimeout);
-    this.unbind();
-    this.remove();
+  remove: function() {
```

Comment:
As I mentioned in some of our earlier conversations, anything bound with listenTo, does *not* need to be unbound in remove(). The Backbone.View.remove will handle that for us.

Some examples of things that *should* go in remove are:
- unbinding of any events not bound with listenTo (listenTo only works for Backbone objects). An example of this is the scroll handler being unbound below. Since it's being bound to a container that lives outside of our view, if we don't unbind the handler, it will still be firing as you scroll in other views. Since it contains a reference to the view we want removed, it will prevent that view from being garbage collected.
- Clearing any timeouts that may have not completed.
...

Notice the last line in this method. It is important that if you implement remove(), you explicitly call the remove of the base view (or whatever view you're extending from) so the inheritance chain is maintained.